### ISSUE TITLE
Bug: `capitalized-comments` configuration accepted by schema/docs but rejected by OXC Zed extension

### ISSUE THREAD

**Issue by tubitubitubi** [oxc-project/zed-oxc#210]
### Summary

The `eslint/capitalized-comments` rule documentation and generated schema indicate that the following configurations are valid:

```json
{
  "rules": {
    "capitalized-comments": [
      "error",
      "always",
      {
        "block": {
          "ignoreConsecutiveComments": false,
          "ignoreInlineComments": false,
          "ignorePattern": "^$"
        },
        "line": {
          "ignoreConsecutiveComments": false,
          "ignoreInlineComments": false,
          "ignorePattern": "^$"
        }
      }
    ]
  }
}
```

However, when this configuration is used with the OXC plugin/extension for Zed, the configuration is treated as invalid. Once this happens, diagnostics stop updating and no further lint errors are reported for other rules or plugins.

### Expected behavior

The configuration should be accepted because:

* The rule documentation explicitly documents:

  * a first option of `"always"` or `"never"`
  * a second options object
* The generated schema introduced in https://github.com/oxc-project/oxc/pull/22984/files appears to allow:

```ts
["error", CapitalizeOption, OptionsJsonDoc]
```

### Actual behavior

The configuration is rejected by the Zed OXC integration, and lint diagnostics stop working afterward.

### Additional observations

The rule only works when configured as:

```json
{
  "rules": {
    "capitalized-comments": "error"
  }
}
```

I also tested:

```json
{
  "rules": {
    "capitalized-comments": ["error", "always"]
  }
}
```

and

```json
{
  "rules": {
    "capitalized-comments": [
      "error",
      {
        "ignorePattern": "^$"
      }
    ]
  }
}
```

but these configurations are also rejected.

### Environment

* Oxlint version: oxlint 1.69.0
* Zed version: 1.6.3
* OXC Zed extension version: 0.4.7

### Possible cause

There may be a mismatch between:

* the rule implementation,
* the generated configuration schema added in [#22984](https://github.com/oxc-project/oxc/pull/22984/files),
* and the configuration validation used by the Zed integration.

The docs and schema indicate that optioned configurations should be valid, but the editor integration appears to reject them.

I will need to delete my account for personal reasons, but the PR is valid and please keep it open.

**Comment by Sysix** [comment]
This is not a bug in oxc-zed, it is fixed in oxlint and will be released in the next version.
